config/web/web.go

```diff
@@ -40,10 +47,27 @@ func (web *Config) ValidateAndSetDefaults() error {
 	} else if web.Port < 0 || web.Port > math.MaxUint16 {
 		return fmt.Errorf("invalid port: value should be between %d and %d", 0, math.MaxUint16)
 	}
+	// Try to load the TLS certificates
+	_, err := web.TLSConfig()
+	if err != nil {
+		return fmt.Errorf("invalid tls config: %s", err)
+	}
 	return nil
 }
 
 // SocketAddress returns the combination of the Address and the Port
 func (web *Config) SocketAddress() string {
 	return fmt.Sprintf("%s:%d", web.Address, web.Port)
 }
+
+// TLSConfig returns a tls.Config object for serving over an encrypted channel
+func (web *Config) TLSConfig() (*tls.Config, error) {
+	if web.CertFile == "" && web.KeyFile == "" {
+		return nil, nil
+	}
+	cer, err := tls.LoadX509KeyPair(web.CertFile, web.KeyFile)
+	if err != nil {
+		return nil, err
+	}
+	return &tls.Config{Certificates: []tls.Certificate{cer}}, nil
+}
```

Comment:
I would rather have a function called `loadTLSConfig()` which only gets applied if `len(web.CertFile) > 0 && len(web.KeyFile) > 0`, and it would set a private field named `tlsConfig` in the `Config` struct which we could then publicly retrieve using the `TLSConfig()` method.

This would allow  us to just read the files once rather than the current twice.

Reply:
Done as suggested. Only that I also had to save the possible error in a private field. Otherwise it would be necessary to attempt the loading twice on error.